Added Ray-Serve Config For LLMs

Blaze-DSP opened the pull request:
Added Example Config For Ray-Serve LLM

kouroshHakha reviewed (commented):
The config looks good to me. (tho I haven't run the config myself)

Blaze-DSP commented:
Should I also add config for autoscaling?

kevin85421 commented:
Chatted with @Blaze-DSP offline

kouroshHakha commented:
What is the plan?

andrewsykim commented on ray-operator/config/samples/ray-service.llm-serve.yaml:
```diff
@@ -0,0 +1,108 @@
+            env:
+            # Hugging Face Token
+            - name: HUGGING_FACE_HUB_TOKEN
+              value: "<HF-TOKEN>"
```
Assuming we'll reference this in guide, please use a secret reference like this for all credentials: https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/vllm/ray-service.vllm.yaml#L40-L45

andrewsykim commented on ray-operator/config/samples/ray-service.llm-serve.yaml:
```diff
@@ -0,0 +1,108 @@
+              value: "<aws-access-key-id>"
+            - name: AWS_SECRET_ACCESS_KEY
+              value: "<aws-secret-access-key>"
+            - name: AWS_DEFAULT_REGION
```
Can we use an example that doesn't require LoRa adapters? Ideally the only credentials we'll require in a guide like this is the hugging face token

kevin85421 commented:
> What is the plan? @kevin85421

Add a doc in Ray repo and make this example simpler (e.g. remove LoRA).

Blaze-DSP commented:
I have updated the ray serve llm config and added doc for it in the ray repo.

PR For Doc.: [ray-serve llm doc](https://github.com/ray-project/ray/pull/52832)

kevin85421 commented on ray-operator/config/samples/ray-service.llm-serve.yaml:
```diff
@@ -0,0 +1,105 @@
+              requests:
+                cpu: 2
+                memory: 4Gi
+            env:
```
does the head Pod need the token?

kevin85421 replied:
does the head Pod need to pull the models from HF?

kevin85421 replied:
@eicherseiji do we need this?

kevin85421 commented on ray-operator/config/samples/ray-service.llm-serve.yaml:
```diff
@@ -0,0 +1,105 @@
+    headGroupSpec:
+      rayStartParams:
+        num-cpus: "0"
+        dashboard-host: "0.0.0.0"
```
```suggestion
```

andrewsykim commented on ray-operator/config/samples/ray-service.llm-serve.yaml:
```diff
@@ -0,0 +1,103 @@
+
+---
+
+apiVersion: v1
```
I suggest just removing this Secret object and instead including an example of how to use `kubectl create secret` in the guide. If you leave this here you can't `kubectl apply -f` the file

kevin85421 replied:
Good point, but I think it’s fine. We can ask users to run a script to replace the placeholder or modify the YAML locally. I just want all the Kubernetes resources to be defined in the YAML.

In addition, I’ll merge the YAML after another round of review. I’ll open follow-up PRs to update the documentation and this YAML.

Blaze-DSP replied:
Great !!! Thanks. Let me know if you need anything else

eicherseiji commented:
Going to give this a shot on my setup in the next week-ish.

eicherseiji reviewed (approved):
Worked great on my setup. Thanks for the PR @Blaze-DSP!

kevin85421 commented:
@eicherseiji could you push to this branch directly to fix CI issues so that I can merge this PR? Thanks!

eicherseiji commented:
- Remove HuggingFace token from head pod since it's not needed there (from my tests).
- Filed https://github.com/ray-project/ray/issues/53641 due to unexpected ray dashboard cluster tab

andrewsykim commented on ray-operator/config/samples/ray-service.llm-serve.yaml:
```diff
@@ -0,0 +1,98 @@
+              limits:
+                cpu: 32
+                memory: 32Gi
+                nvidia.com/gpu: "4"
```
Ah I noticed that tensor parallelism is not set, so it must only be using 1 GPU,  I suggest updating this example to only request 1 GPU for each worker
